src/components.ts

```diff
@@ -1,19 +1,45 @@
-import { createDotEnvConfigComponent } from "@well-known-components/env-config-provider"
-import { createServerComponent, createStatusCheckComponent } from "@well-known-components/http-server"
-import { createLogComponent } from "@well-known-components/logger"
-import { createFetchComponent } from "./adapters/fetch"
-import { createMetricsComponent, instrumentHttpServerWithMetrics } from "@well-known-components/metrics"
-import { AppComponents, GlobalContext } from "./types"
-import { metricDeclarations } from "./metrics"
+import {
+  createAwsS3BasedFileSystemContentStorage,
+  createFolderBasedFileSystemContentStorage,
+  createFsComponent
+} from '@dcl/catalyst-storage'
+import { createDotEnvConfigComponent } from '@well-known-components/env-config-provider'
+import { createServerComponent, createStatusCheckComponent } from '@well-known-components/http-server'
+import { createLogComponent } from '@well-known-components/logger'
+import { createMetricsComponent, instrumentHttpServerWithMetrics } from '@well-known-components/metrics'
+import * as os from 'os'
+import { createFetchComponent } from './adapters/fetch'
+import { createFileManager } from './adapters/fileSystem'
+import createAssetConverter from './logic/asset-converter'
+import createAssetRetriever from './logic/assetRetriever'
+import { metricDeclarations } from './metrics'
+import { AppComponents, GlobalContext } from './types'
+import createCommandTrigger from './adapters/commandLine'
 
 // Initialize all the components of the app
 export async function initComponents(): Promise<AppComponents> {
-  const config = await createDotEnvConfigComponent({ path: [".env.default", ".env"] })
+  const config = await createDotEnvConfigComponent({ path: ['.env.default', '.env'] })
   const metrics = await createMetricsComponent(metricDeclarations, { config })
   const logs = await createLogComponent({ metrics })
   const server = await createServerComponent<GlobalContext>({ config, logs }, {})
   const statusChecks = await createStatusCheckComponent({ server, config })
   const fetch = await createFetchComponent()
+  const fs = createFsComponent()
+
+  const bucket = await config.getString('BUCKET')
+  const storageDirectory = (await config.getString('STORAGE_DIRECTORY')) || `${os.tmpdir()}`
+
+  const bucketStorage = bucket
+    ? await createAwsS3BasedFileSystemContentStorage({ fs, config }, bucket)
+    : await createFolderBasedFileSystemContentStorage({ fs }, storageDirectory)
+
+  const storages = {
+    local: createFileManager(storageDirectory),
+    bucket: bucketStorage
+  }
+
+  const assetConverter = createAssetConverter(createCommandTrigger())
+  const assetRetriever = await createAssetRetriever((await config.getString('CONTENT_URL')) || '')
```

Comment:
I'd suggest to create the component passing the dependencies and moving the `config` reader inside of the `createAssetRetriever` function. It also re-creates the fetch component, which should be passed on in the function apply
```suggestion
  const assetRetriever = await createAssetRetriever({ config, fetch })
```